File: SFDataApplicationTestC++/shadow/system/data/objects/SFFloatArray.cpp


#include "SFFloatArray.h"



namespace sf{
	SFFloatArray::SFFloatArray(int floatValuesSize) {
		this->floatValues = new float[floatValuesSize];
        this->n=floatValuesSize;
	}

	SFFloatArray::~SFFloatArray(){
        delete floatValues;
    }

	float* SFFloatArray::getFloatValues() {
		return floatValues;
	}

	void SFFloatArray::setFloatValues(float* FloatValues) {
		delete floatValues;
		this->floatValues = FloatValues;
	}


	void SFFloatArray::readFromStream(SFInputStream* stream) {
		delete floatValues;
		n=stream->readShort();
		floatValues=stream->readFloats(n);
	}

	SFFloatArray* SFFloatArray::clone(){
		return new SFFloatArray(n);
	}
}

File: ShadowFrameworkLite20/src/shadow/system/data/objects/SFFloatArray.java
package shadow.system.data.objects;

import shadow.system.data.SFCharsetObject;
import shadow.system.data.SFInputStream;
import shadow.system.data.SFOutputStream;

public class SFFloatArray extends SFPrimitiveType  implements SFCharsetObject{

	private float[] floatValues;
	
	public SFFloatArray(int floatValuesSize) {
		super();
		this.floatValues = new float[floatValuesSize];
	}

	
	public float[] getFloatValues() {
		return floatValues;
	}

	public void setFloatValues(float[] FloatValues) {
		this.floatValues = FloatValues;
	}

	@Override
	public void readFromStream(SFInputStream stream) {
		int n=stream.readShort();
		floatValues=stream.readFloats(n);
	}
	
	@Override
	public void writeOnStream(SFOutputStream stream) {
		stream.writeShort((short)floatValues.length);
		stream.writeFloats(floatValues);
	}
	
	@Override
	public SFFloatArray clone(){
		return new SFFloatArray(floatValues.length);
	}
	

	@Override
	public void setStringValue(String value) {
		SFCharsetObjectUtils.readFloats(floatValues, value, this.getClass().getSimpleName());
	}
	
	@Override
	public String toStringValue() {
		return SFCharsetObjectUtils.writeFloats(floatValues);
	}
}
